README.md

```diff
@@ -52,27 +52,82 @@ the provided `OpenTracing::Tracer`
 If you prefer direct control to singletons, manage ownership of the
 `Tracer` implementation explicitly.
 
-### Starting an empty trace by creating a "root span"
+### Scopes and within-process propagation
 
-It's always possible to create a "root" `Span` with no parent or other causal
-reference.
+For any thread, at most one `Span` may be "active". Of course there may be many
+other `Spans` involved with the thread which are (a) started, (b) not finished,
+and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child
+`Span`, or otherwise off of the critical path.
+
+It's inconvenient to pass an active `Span` from function to function manually,
+so OpenTracing requires that every `Tracer` contains a `ScopeManager` that
+grants access to the active `Span` through a `Scope`. Any `Span` may be
+transferred to another callback or thread, but not `Scope`.
+
+#### Accessing the active Span through Scope
 
 ```ruby
-span = OpenTracing.start_span("operation_name")
-span.finish
+# Access to the active span is straightforward.
+scope = OpenTracing.scope_manager.active
+if scope
+  scope.span.set_tag('...', '...')
+end
 ```
 
-This example will start a span on the global tracer (initialized above). If
-you are managing your own tracer you'll need to call `start_span` on your
-tracer.
+### Starting a new Span
+
+The common case starts a `Scope` that's automatically registered for
+intra-process propagation via `ScopeManager`.
 
-### Creating a (child) Span given an existing (parent) Span
+Note that `start_active_span('...')` automatically finishes the span on
+`Scope#close` (`start_active_span('...', finish_on_close: false)` does not
+finish it, in contrast).
 
 ```ruby
-span = OpenTracing.start_span("parent")
-child = OpenTracing.start_span("child", child_of: span)
-child.finish
-span.finish
+# Automatic activation of the Span.
+# By default the active span will be finished when the returned scope is closed.
+# This can be controlled by passing finish_on_close parameter to
+# start_active_span
+scope = OpenTracing.start_active_span('operation_name')
+# Do things.
+
+# Block form of start_active_span
+# start_active_span optionally accepts a block. If a block is passed to
+# start_active_span it will yield the newly created scope. The scope will
+# be closed and its associated span will be finished unless
+# finish_on_close: false is passed to start_active_span.
+OpenTracing.start_active_span('operation_name') do |scope|
+# Do things.
+end
+
+# Manual activation of the Span.
+# Spans can be managed manually. This is equivalent to the more concise examples
+# above.
+span = OpenTracing.start_span('operation_name')
+OpenTracing.scope_manager.activate(span, true)
```

Comment:
this should be `OpenTracing.scope_manager.activate(span, finish_on_close: true)`

Reply:
Right you are. Thanks for catching that!